### PR TITLE
docs: Update Linux distros, add WSL, and archs to Quickstart

### PR DESCRIPTION
**Pull request by sirredbeard**
## Description

This PR updates the list of supported Linux distros, adds WSL as a platform, and adds Arm and PowerPC architectures to the Quickstart for Model Developers

## Test Plan

These updates to the documentation do not require testing or reference changes/documentation that have already been tested, such as #8658

### CODE REVIEW

File: docs/tutorials/quickstart-mdldev.rst [hunk 1]
```diff
@@ -31,12 +31,17 @@ information about each, visit the :ref:`Experiment Configuration Reference
 Software
 ========
 
--  Determined *agent* and *master* nodes must be configured with Ubuntu 16.04 or higher, CentOS 7,
-   or macOS 10.13 or higher.
+-  Determined *agent* and *master* nodes must be configured with the following operating systems:
+
+   - Ubuntu 20.04 or higher
```
Comment (tara-det-ai) [inline docs/tutorials/quickstart-mdldev.rst:36]: - Ubuntu 20.04 or later

File: docs/tutorials/quickstart-mdldev.rst [hunk 2]
```diff
@@ -31,12 +31,17 @@ information about each, visit the :ref:`Experiment Configuration Reference
...
+   - Enterprise Linux (e.g., AlmaLinux, Oracle Linux, Red Hat Enterprise Linux, or Rocky Linux) 7 or higher
+   - Windows Subsystem for Linux (WSL) 2 on Windows 10 version 1903 or higher, with a supported Linux distribution
+     installed
+   - macOS 10.13 or higher
```
Comment (tara-det-ai) [inline docs/tutorials/quickstart-mdldev.rst:40]: - macOS 10.13 or later

File: docs/tutorials/quickstart-mdldev.rst [hunk 3]
```diff
@@ -31,12 +31,17 @@ information about each, visit the :ref:`Experiment Configuration Reference
...
 
 -  Agent nodes must have Docker installed.
 
--  To run jobs with GPUs, install NVIDIA drivers, version 384.81 or higher, on each agent. The
+-  To run jobs with GPUs, NVIDIA drivers, version 384.81 or higher, are required on each agent. The
```
Comment (tara-det-ai) [inline docs/tutorials/quickstart-mdldev.rst:44]: To run jobs that use GPUs, you need NVIDIA drivers with a version of 384.81 or later installed on each agent. These drivers can be installed independently, but you don't need to install the entire CUDA toolkit.

**Review by tara-det-ai — APPROVED** [review]
requested edits

**Comment by sirredbeard** [comment]
edits complete

**Review by tara-det-ai — APPROVED** [review]
LGTM

**Pull request merged**